src/tree/v2/application/ResourceGroupsTreeDataProvider.ts

```diff
@@ -0,0 +1,155 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { AzExtServiceClientCredentials, nonNullProp } from '@microsoft/vscode-azext-utils';
+import { AzureExtensionApiProvider } from '@microsoft/vscode-azext-utils/api';
+import * as vscode from 'vscode';
+import { ApplicationResourceProviderManager } from '../../../api/v2/ApplicationResourceProviderManager';
+import { ResourceModelBase } from '../../../api/v2/v2AzureResourcesApi';
+import { localize } from '../../../utils/localize';
+import { ApplicationResourceGroupingManager } from './ApplicationResourceGroupingManager';
+import { AzureAccountExtensionApi } from '../azure-account.api';
+import { GenericItem } from '../GenericItem';
+import { ResourceGroupsItem } from '../ResourceGroupsItem';
+import { ResourceGroupsItemCache } from '../ResourceGroupsItemCache';
+import { ResourceTreeDataProviderBase } from '../ResourceTreeDataProviderBase';
+import { SubscriptionItem } from './SubscriptionItem';
+
+export class ResourceGroupsTreeDataProvider extends ResourceTreeDataProviderBase {
```

Comment:
Rename this to `ApplicationTreeDataProvider` for consistency? Since the other TreeDataProvider is named `WorkspaceTreeDataProvider`.

Reply:
Sounds good; I've renamed them to `ApplicationResourceTreeDataProvider` and `WorkspaceResourceTreeDataProvider`.